src/components/common/CopyButton/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { faCopy } from '@fortawesome/free-regular-svg-icons'
+import { faCheck } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import CopyToClipboard from 'react-copy-to-clipboard'
+
+import { DISPLAY_TEXT_COPIED_CHECK } from 'apps/explorer/const'
+
+const Icon = styled(FontAwesomeIcon)<{ copied: boolean }>`
+  color: ${({ theme, copied }): string => (copied ? theme.green1 : theme.icon)};
+  cursor: ${({ copied }): string => (copied ? 'reset' : 'pointer')};
+`
+
+export type Props = { text: string }
+
+/**
+ * Simple CopyButton component.
+ *
+ * Takes in the `text` to be copied when button is clicked
+ * Displays the copy icon with theme based color
+ * When clicked, displays a green check for DISPLAY_TEXT_COPIED_CHECK seconds,
+ * then is back to original copy icon
+ */
+export function CopyButton(props: Props): JSX.Element {
+  const { text } = props
```

Comment:
not a fan of destructuring in signature are you :joy: :D

Reply:
Nope, I like the signature cleaner :)

Reply:
I'm with @alfetopito in this one, with one param I find it okeish, but I also don't like this tough-guy-lines with declarations, arrow functions, ternary operators and || && ones. Yes, when you write them u feel u are a hacker and is clear, but the next one reading that, won't have such an easy time

1 thing per line as a general rule, sounds good to me. Of Course with exceptions